### DISCUSSION TITLE
Mobile development workflow

### DISCUSSION THREAD

**Discussion by markusamshove** [AvaloniaUI/Avalonia#12449] [Q&A]
Heya,

I'd like to start a discussion about your development workflow and experience with Avalonia for mobile.

I've tested it myself on Mac and would like to use it for an upcoming app, but I found the experience about building not so great.

## XAML Preview

I'm using Rider and the Avalonia plug-in for it, however the XAML preview only features desktop and browser. To workaround this, I've set the desktop resolution of the main view to resemble a smartphone.

## Hot Reload/Rebuild

I'm finding myself to always having to rebuild and redeploy the App to have it up to date with the source code. That's a bummer compared to the experience with e.g. Flutter.
I haven't tested the live reload nuget package yet, as it states that changes from referenced projects aren't accounted for. Since the cross platform solution setup is a shared project with platform specific projects, I don't think it will work.

How do you develop Avalonia on mobile and how does your feedback loop look like while developing?

**Comment by maxkatz6** [comment]
> I'm using Rider and the Avalonia plug-in for it, however the XAML preview only features desktop and browser. To workaround this, I've set the desktop resolution of the main view to resemble a smartphone.

Rider/VS previewer do not feature onlg desktop or browser. It doesn't feature any platform specifically but previewer. What's your specific issue there in mobile context? 
As for resolution, you can set Design.Width and Design.Height properties in XAML root element like UserControl. These properties will be used in the previewer. 

But yeah, it would be nice to have a resolution selector like Chrome has.

**Reply by markusamshove** [reply]
I guess Support was a hard word, sorry for the confusion. The previewed drop-down only offers the desktop and browser project.
The design height and width is what I'm currently using

**Reply by kekekeks** [reply]
Previewer actually runs your application in background with a special platform backend and without lifetime. That allows us to provide 100% accurate preview that properly accounts for any customizations you do in App.xaml.

Unfortunately it's not possible to run mobile TFMs on desktop, so a desktop launcher project is needed. That's one of the reasons why the template has a common net6.0-targeted project with separate desktop/mobile launchers.

**Reply by maxkatz6** [reply]
> The previewed drop-down only offers the desktop and browser project.

That's just your project names. And previewer needs a project to run.
